tests/unit/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.spec.jsx

```diff
@@ -0,0 +1,137 @@
+import { fireEvent, screen, waitFor } from '@testing-library/react'
+
+import CoursesFiltersDrawer from '~/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer'
+
+import { renderWithProviders } from '~tests/test-utils'
+import { mockAxiosClient, selectOption } from '~tests/test-utils'
+import { URLs } from '~/constants/request'
+import { ProficiencyLevelEnum } from '~/types'
+
+const mockUpdateFiltersInQuery = vi.fn()
+const mockResetFilters = vi.fn()
+const mockOnClose = vi.fn()
+
+const defaultFilters = {
+  category: '',
+  subject: '',
+  proficiencyLevel: [],
+  title: ''
+}
+const filterWithCategory = {
+  category: '1',
+  subject: '',
+  proficiencyLevel: [],
+  title: ''
+}
+
+const filterActions = {
+  updateFiltersInQuery: mockUpdateFiltersInQuery,
+  resetFilters: mockResetFilters
+}
+
+const mockCategories = [
+  { _id: '1', name: 'Mathematics' },
+  { _id: '2', name: 'Science' }
+]
+
+const mockSubjects = [{ _id: '1', name: 'Algebra' }]
+
+const setup = async (filters) => {
+  mockAxiosClient.onGet(URLs.categories.getNames).reply(200, mockCategories)
+  mockAxiosClient
+    .onGet(`${URLs.categories.get}/1${URLs.subjects.getNames}`)
+    .reply(200, mockSubjects)
+
+  await waitFor(() => {
+    renderWithProviders(
+      <CoursesFiltersDrawer
+        additionalParams={{}}
+        filterActions={filterActions}
+        filters={filters}
+        isOpen
+        onClose={mockOnClose}
+      />
+    )
+  })
```

Comment:
I think it should work without `waitFor`

Reply:
I think this line is required, withour `waitFor` we receiving this warnings in console:

![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/118522791/93eceaaf-ead6-4388-a647-bbc2c2450047)

Reply:
I noticed that a lot of tests don't use waitFor. Maybe we should create a non-urgent issue for it?

Reply:
i think we need to discuss about it on our meeting